test/integration/controlplane/transformation/transformation_testcase.go

```diff
@@ -134,17 +139,39 @@ func (e *transformTest) run(unSealSecretFunc unSealSecret, expectedEnvelopePrefi
 		e.logger.Errorf("failed to unseal secret: %v", err)
 		return
 	}
-	if !strings.Contains(string(v), secretVal) {
-		e.logger.Errorf("expected %q after decryption, but got %q", secretVal, string(v))
+	if e.kind == "secrets" {
+		if !strings.Contains(string(v), secretVal) {
+			e.logger.Errorf("expected %q after decryption, but got %q", secretVal, string(v))
+		}
+	} else {
+		if !strings.Contains(string(v), e.name) {
+			e.logger.Errorf("expected %q after decryption, but got %q", e.name, string(v))
+		}
 	}
 
-	// Secrets should be un-enveloped on direct reads from Kube API Server.
-	s, err := e.restClient.CoreV1().Secrets(testNamespace).Get(context.TODO(), testSecret, metav1.GetOptions{})
-	if err != nil {
-		e.logger.Errorf("failed to get Secret from %s, err: %v", testNamespace, err)
-	}
-	if secretVal != string(s.Data[secretKey]) {
-		e.logger.Errorf("expected %s from KubeAPI, but got %s", secretVal, string(s.Data[secretKey]))
+	// Data should be un-enveloped on direct reads from Kube API Server.
+	if e.kind == "secrets" {
+		s, err := e.restClient.CoreV1().Secrets(testNamespace).Get(context.TODO(), testSecret, metav1.GetOptions{})
+		if err != nil {
+			e.logger.Errorf("failed to get Secret from %s, err: %v", testNamespace, err)
+		}
+		if secretVal != string(s.Data[secretKey]) {
+			e.logger.Errorf("expected %s from KubeAPI, but got %s", secretVal, string(s.Data[secretKey]))
+		}
+	} else {
+		e.logger.Logf("Get object with dynamic client")
+		dynamicClient, err := dynamic.NewForConfig(e.kubeAPIServer.ClientConfig)
+		if err != nil {
+			e.logger.Fatalf("Unexpected error: %v", err)
+		}
+		fooResource := schema.GroupVersionResource{Group: e.group, Version: "v1", Resource: e.kind}
+		obj, err := dynamicClient.Resource(fooResource).Get(context.TODO(), e.name, metav1.GetOptions{})
+		if err != nil {
+			e.logger.Errorf("Failed to get test instance: %v, name: %s", err, e.name)
+		}
+		if obj.GetName() != e.name {
+			e.logger.Errorf("expected %s from KubeAPI, but got %s", e.name, obj.GetName())
+		}
```

Comment:
Check that the GVR + namespace + name match what you expect (since those 5 things uniquely identify an object).